src/lib/ibcclient.ts

```diff
@@ -99,15 +122,114 @@ export class IbcClient {
     return this.tm.commit(height);
   }
 
+  public async getSignedHeader(height?: number): Promise<SignedHeader> {
+    const { header: rpcHeader, commit: rpcCommit } = await this.getCommit(
+      height
+    );
+    const header = Header.fromPartial({
+      ...rpcHeader,
+      version: {
+        block: new Long(rpcHeader.version.block),
+      },
+      height: new Long(rpcHeader.height),
+      time: timestampFromDateNanos(rpcHeader.time),
+      lastBlockId: {
+        hash: rpcHeader.lastBlockId.hash,
+        partSetHeader: rpcHeader.lastBlockId.parts,
+      },
+    });
+
+    const signatures = rpcCommit.signatures.map((sig) => ({
+      ...sig,
+      timestamp: sig.timestamp && timestampFromDateNanos(sig.timestamp),
+      blockIdFlag: blockIDFlagFromJSON(sig.blockIdFlag),
+    }));
+    const commit = Commit.fromPartial({
+      height: new Long(rpcCommit.height),
+      round: rpcCommit.round,
+      blockId: {
+        hash: rpcCommit.blockId.hash,
+        partSetHeader: rpcCommit.blockId.parts,
+      },
+      signatures,
+    });
+    // For the vote sign bytes, it checks (from the commit):
+    //   Height, Round, BlockId, TimeStamp, ChainID
+
+    return { header, commit };
+  }
+
+  public async getValidatorSet(height: number): Promise<ValidatorSet> {
+    // TODO: use header not commit
+    // we need to query the header to find out who the proposer was, and pull them out
+    const { proposerAddress } = (await this.getCommit(height)).header;
+    const validators = await this.tm.validators(height);
+    const mappedValidators = validators.validators.map((val) => ({
+      address: val.address,
+      // TODO: map to handle secp as well (check val.pubkey.type)
+      pubKey: val.pubkey
+        ? {
+            ed25519: val.pubkey.data,
+          }
+        : undefined,
+      votingPower: new Long(val.votingPower),
+      proposerPriority: val.proposerPriority
+        ? new Long(val.proposerPriority)
+        : undefined,
+    }));
+    const totalPower = validators.validators.reduce(
+      (x, v) => x + v.votingPower,
+      0
+    );
+    const proposer = mappedValidators.find(
+      (val) => toHex(val.address) === toHex(proposerAddress)
```

Comment:
You can use `import { arrayContentEquals } from "@cosmjs/utils";` to avoid the hex conversion

```suggestion
      (val) => arrayContentEquals(val.address, proposerAddress)
```